#117 Change how trees are assigned into PFTs

arne-exe opened the pull request:
This PR will provide a set of scripts to implement a new way to assign trees into PFTs:
- add PFT species classification script based on Dobert et al., 2017 data
- generate 2 new csv files (to be uploaded in Globus)
- add new cohort distribution script, that takes the Dobert species classification and applies it to the SAFE tree census data
- generate new cohort distribution csv file (also to be uploaded in Globus)

Note to self: also upload updated output files for base cohort distribution to Globus (see files changed: 4x)

arne-exe committed:
Created PFT species classification using Dobert data

arne-exe commented:
Hi @davidorme, it's the first time pushing a commit from a new laptop, I think I set up everything except for the pre-commit tool. Can you please remind me how this is done or where I can read how to do this?

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

arne-exe commented:
> Hi @davidorme, it's the first time pushing a commit from a new laptop, I think I set up everything except for the pre-commit tool. Can you please remind me how this is done or where I can read how to do this?

Think I found it: https://virtual-ecosystem.readthedocs.io/en/latest/development/contributing/overview.html#installing-and-using-pre-commit

arne-exe commented:
> Hi @davidorme, it's the first time pushing a commit from a new laptop, I think I set up everything except for the pre-commit tool. Can you please remind me how this is done or where I can read how to do this?
> 
> Note to self: Think I found it: https://virtual-ecosystem.readthedocs.io/en/latest/development/contributing/overview.html#installing-and-using-pre-commit
> 
> * install poetry
> * install pre-commit
> * add R to path
> * make sure Rtools is installed
> * add Rtools to path

```
$ git commit -m "Recommit after installing poetry and precommit"
[INFO] Installing environment for https://github.com/lorenzwalthert/precommit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('C:\\Program Files\\R\\R-4.5.1\\bin\\Rscript.EXE', '--vanilla', 'C:\\Users\\User\\AppData\\Local\\Temp\\tmpbxrpzqmr\\script.R')
return code: 1
stdout:
    # Bootstrapping renv 1.0.11 --------------------------------------------------
    - Downloading renv ... OK
    - Installing renv  ... OK

    - One or more packages recorded in the lockfile are not installed.
    - Use `renv::status()` for more details.
    The following package(s) will be updated:

    # RSPM -----------------------------------------------------------------------
    - backports      [* -> 1.5.0]
    - brew           [* -> 1.0-10]
    - callr          [* -> 3.7.6]
    - cli            [* -> 3.6.3]
    - commonmark     [* -> 1.9.2]
    - cpp11          [* -> 0.5.1]
    - crayon         [* -> 1.5.3]
    - curl           [* -> 6.2.0]
    - cyclocomp      [* -> 1.1.1]
    - desc           [* -> 1.4.3]
    - digest         [* -> 0.6.37]
    - docopt         [* -> 0.7.1]
    - evaluate       [* -> 1.0.3]
    - fs             [* -> 1.6.5]
    - glue           [* -> 1.8.0]
    - here           [* -> 1.0.1]
    - highr          [* -> 0.11]
    - hunspell       [* -> 3.0.5]
    - jsonlite       [* -> 1.8.9]
    - jsonvalidate   [* -> 1.3.2]
    - knitr          [* -> 1.49]
    - lazyeval       [* -> 0.2.2]
    - lifecycle      [* -> 1.0.4]
    - lintr          [* -> 3.1.2]
    - magrittr       [* -> 2.0.3]
    - pkgbuild       [* -> 1.4.6]
    - pkgload        [* -> 1.4.0]
    - processx       [* -> 3.8.5]
    - ps             [* -> 1.8.1]
    - purrr          [* -> 1.0.2]
    - R.cache        [* -> 0.16.0]
    - R.methodsS3    [* -> 1.8.2]
    - R.oo           [* -> 1.27.0]
    - R.utils        [* -> 2.12.3]
    - R6             [* -> 2.5.1]
    - Rcpp           [* -> 1.0.14]
    - remotes        [* -> 2.5.0]
    - rex            [* -> 1.2.1]
    - rlang          [* -> 1.1.5]
    - roxygen2       [* -> 7.3.2]
    - rprojroot      [* -> 2.0.4]
    - spelling       [* -> 2.3.1]
    - stringi        [* -> 1.8.4]
    - stringr        [* -> 1.5.1]
    - styler         [* -> 1.10.3]
    - V8             [* -> 6.0.0]
    - vctrs          [* -> 0.6.5]
    - withr          [* -> 3.0.2]
    - xfun           [* -> 0.50]
    - xml2           [* -> 1.3.6]
    - xmlparsedata   [* -> 1.0.5]
    - yaml           [* -> 2.3.10]

    # Downloading packages -------------------------------------------------------
    - Downloading V8 from RSPM ...                  OK [file is up to date]
    - Downloading jsonlite from RSPM ...            OK [file is up to date]
    - Downloading curl from RSPM ...                OK [file is up to date]
    - Downloading backports from RSPM ...           OK [file is up to date]
    - Downloading brew from RSPM ...                OK [file is up to date]
    - Downloading callr from RSPM ...               OK [file is up to date]
    - Downloading processx from RSPM ...            OK [file is up to date]
    - Downloading ps from RSPM ...                  OK [file is up to date]
    - Downloading cli from RSPM ...                 OK [file is up to date]
    - Downloading commonmark from RSPM ...          OK [file is up to date]
    - Downloading cpp11 from RSPM ...               OK [file is up to date]
    - Downloading crayon from RSPM ...              OK [file is up to date]
    - Downloading cyclocomp from RSPM ...           OK [file is up to date]
    - Downloading desc from RSPM ...                OK [file is up to date]
    - Downloading remotes from RSPM ...             OK [file is up to date]
    - Downloading withr from RSPM ...               OK [file is up to date]
    - Downloading docopt from RSPM ...              OK [file is up to date]
    - Downloading evaluate from RSPM ...            OK [file is up to date]
    - Downloading fs from RSPM ...                  OK [file is up to date]
    - Downloading glue from RSPM ...                OK [file is up to date]
    - Downloading here from RSPM ...                OK [file is up to date]
    - Downloading rprojroot from RSPM ...           OK [file is up to date]
    - Downloading highr from RSPM ...               OK [file is up to date]
    - Downloading xfun from RSPM ...                OK [file is up to date]
    - Downloading hunspell from RSPM ...            OK [file is up to date]
    - Downloading jsonvalidate from RSPM ...        OK [file is up to date]
    - Downloading knitr from RSPM ...               OK [file is up to date]
    - Downloading yaml from RSPM ...                OK [file is up to date]
    - Downloading lazyeval from RSPM ...            OK [file is up to date]
    - Downloading lifecycle from RSPM ...           OK [file is up to date]
    - Downloading rlang from RSPM ...               OK [file is up to date]
    - Downloading lintr from RSPM ...               OK [file is up to date]
    - Downloading rex from RSPM ...                 OK [file is up to date]
    - Downloading xml2 from RSPM ...                OK [file is up to date]
    - Downloading xmlparsedata from RSPM ...        OK [file is up to date]
    - Downloading magrittr from RSPM ...            OK [file is up to date]
    - Downloading pkgbuild from RSPM ...            OK [file is up to date]
    - Downloading pkgload from RSPM ...             OK [file is up to date]
    - Downloading purrr from RSPM ...               OK [file is up to date]
    - Downloading vctrs from RSPM ...               OK [file is up to date]
    - Downloading roxygen2 from RSPM ...            OK [file is up to date]
    - Downloading stringi from RSPM ...             OK [file is up to date]
    - Downloading stringr from RSPM ...             OK [file is up to date]
    - Downloading spelling from RSPM ...            OK [file is up to date]
    - Downloading styler from RSPM ...              OK [file is up to date]
    Successfully downloaded 45 packages in 56 seconds.

    # Installing packages --------------------------------------------------------
    - Installing R.methodsS3 ...                    OK [linked from cache]
    - Installing R.oo ...                           OK [linked from cache]
    - Installing R.utils ...                        OK [linked from cache]
    - Installing digest ...                         OK [linked from cache]
    - Installing R.cache ...                        OK [linked from cache]
    - Installing R6 ...                             OK [linked from cache]
    - Installing Rcpp ...                           OK [linked from cache]
    - Installing jsonlite ...                       FAILED
    Traceback (most recent calls last):
    13: renv::restore()
    12: renv_restore_run_actions(project, diff, current, lockfile, rebuild)
    11: renv_install_impl(records)
    10: renv_install_staged(records)
     9: renv_install_default(records)
     8: handler(package, renv_install_package(record))
     7: renv_install_package(record)
     6: withCallingHandlers(renv_install_package_impl(record), error = function(e) writef("FAILED"))
     5: renv_install_package_impl(record)
     4: r_cmd_install(package, path)
     3: r_exec_error(package, output, "install", status)
     2: abort(all)
     1: stop(fallback)
stderr:
    Error: Error installing package 'jsonlite':
    ====================================

    * installing *source* package 'jsonlite' ...
    ** this is package 'jsonlite' version '1.8.9'
    ** package 'jsonlite' successfully unpacked and MD5 sums checked
    ** using staged installation
    ** libs
    using C compiler: 'gcc.exe (GCC) 14.2.0'
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -Iyajl/api    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"    -D__USE_MINGW_ANSI_STDIO   -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c base64.c -o base64.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -Iyajl/api    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"    -D__USE_MINGW_ANSI_STDIO   -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c collapse_array.c -o collapse_array.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -Iyajl/api    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"    -D__USE_MINGW_ANSI_STDIO   -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c collapse_object.c -o collapse_object.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -Iyajl/api    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"    -D__USE_MINGW_ANSI_STDIO   -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c collapse_pretty.c -o collapse_pretty.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -Iyajl/api    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"    -D__USE_MINGW_ANSI_STDIO   -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c escape_chars.c -o escape_chars.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -Iyajl/api    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"    -D__USE_MINGW_ANSI_STDIO   -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c integer64_to_na.c -o integer64_to_na.o
    In file included from modp_numtoa.h:34,
                     from integer64_to_na.c:2:
    modp_stdint.h:37:23: error: two or more data types in declaration specifiers
       37 | typedef unsigned char bool;
          |                       ^~~~
    modp_stdint.h:37:1: warning: useless type name in empty declaration
       37 | typedef unsigned char bool;
          | ^~~~~~~
    make: *** [C:/PROGRA~1/R/R-45~1.1/etc/x64/Makeconf:289: integer64_to_na.o] Error 1
    ERROR: compilation failed for package 'jsonlite'
    * removing 'C:/Users/User/.cache/pre-commit/repog6trpveh/renv-default/renv/staging/1/jsonlite'
    install of package 'jsonlite' failed [error code 1]
    Execution halted
Check the log at C:\Users\User\.cache\pre-commit\pre-commit.log
```
Hi @davidorme, I'm stuck here, do you have any suggestions on how to fix this?

davidorme commented:
@arne-exe We also have the `ve_data_science` website here:

https://imperialcollegelondon.github.io/ve_data_science/getting_started/

Although that page is a little out of date and doesn't say anything much different from the VE site. We do need to improve this though - some people are relying on the GitHub CI to catch `pre-commit` issues and it is better to trap them locally!

On this _specific_ issue... I'm not sure but I think the first thing to do is to update the `pre-commit` chain. I'm going to put in a PR for that now.

davidorme committed:
Merge branch 'main' into 117-change-how-trees-are-assigned-into-pfts

arne-exe commented:
Hi @davidorme, thanks for looking into this. I still can't get it to work, even after the following:
- installed poetry (and pre-commit)
- set the correct virtual environment
- cleaned the pre-commit cache
- installed R and Rtools (and added them to system path)
- installed the "digest package"

I've attached a copy of the error below:
```
$ poetry run pre-commit run --all-files
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Initializing environment for https://github.com/igorshubovych/markdownlint-cli.
[INFO] Initializing environment for https://github.com/lorenzwalthert/precommit.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/igorshubovych/markdownlint-cli.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/lorenzwalthert/precommit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('C:\\Program Files\\R\\R-4.5.1\\bin\\Rscript.EXE', '--vanilla', 'C:\\Users\\User\\AppData\\Local\\Temp\\tmphvy19zw8\\script.R')
return code: 1
stdout:
    # Bootstrapping renv 1.0.5 ---------------------------------------------------
    - Downloading renv ... OK
    - Installing renv  ... OK

    - One or more packages recorded in the lockfile are not installed.
    - Use `renv::status()` for more details.
    The following package(s) will be updated:

    # RSPM -----------------------------------------------------------------------
    - backports      [* -> 1.5.0]
    - brew           [* -> 1.0-10]
    - callr          [* -> 3.7.6]
    - cli            [* -> 3.6.3]
    - commonmark     [* -> 1.9.1]
    - cpp11          [* -> 0.4.7]
    - crayon         [* -> 1.5.3]
    - cyclocomp      [* -> 1.1.1]
    - desc           [* -> 1.4.3]
    - digest         [* -> 0.6.36]
    - docopt         [* -> 0.7.1]
    - evaluate       [* -> 0.24.0]
    - fs             [* -> 1.6.4]
    - glue           [* -> 1.7.0]
    - here           [* -> 1.0.1]
    - highr          [* -> 0.11]
    - hunspell       [* -> 3.0.3]
    - knitr          [* -> 1.48]
    - lazyeval       [* -> 0.2.2]
    - lifecycle      [* -> 1.0.4]
    - lintr          [* -> 3.1.2]
    - magrittr       [* -> 2.0.3]
    - pkgbuild       [* -> 1.4.4]
    - pkgload        [* -> 1.4.0]
    - processx       [* -> 3.8.4]
    - ps             [* -> 1.7.7]
    - purrr          [* -> 1.0.2]
    - R.cache        [* -> 0.16.0]
    - R.methodsS3    [* -> 1.8.2]
    - R.oo           [* -> 1.26.0]
    - R.utils        [* -> 2.12.3]
    - R6             [* -> 2.5.1]
    - Rcpp           [* -> 1.0.12]
    - remotes        [* -> 2.5.0]
    - rex            [* -> 1.2.1]
    - rlang          [* -> 1.1.4]
    - roxygen2       [* -> 7.3.2]
    - rprojroot      [* -> 2.0.4]
    - spelling       [* -> 2.3.0]
    - stringi        [* -> 1.8.4]
    - stringr        [* -> 1.5.1]
    - styler         [* -> 1.10.3]
    - vctrs          [* -> 0.6.5]
    - withr          [* -> 3.0.0]
    - xfun           [* -> 0.45]
    - xml2           [* -> 1.3.6]
    - xmlparsedata   [* -> 1.0.5]
    - yaml           [* -> 2.3.9]

    # Downloading packages -------------------------------------------------------
    - Downloading R.cache from RSPM ...             OK [file is up to date]
    - Downloading digest from RSPM ...              OK [file is up to date]
    - Downloading R6 from RSPM ...                  OK [file is up to date]
    - Downloading Rcpp from RSPM ...                OK [file is up to date]
    - Downloading backports from RSPM ...           OK [file is up to date]
    - Downloading brew from RSPM ...                OK [file is up to date]
    - Downloading callr from RSPM ...               OK [file is up to date]
    - Downloading processx from RSPM ...            OK [file is up to date]
    - Downloading ps from RSPM ...                  OK [file is up to date]
    - Downloading cli from RSPM ...                 OK [file is up to date]
    - Downloading commonmark from RSPM ...          OK [file is up to date]
    - Downloading cpp11 from RSPM ...               OK [file is up to date]
    - Downloading crayon from RSPM ...              OK [file is up to date]
    - Downloading cyclocomp from RSPM ...           OK [file is up to date]
    - Downloading desc from RSPM ...                OK [file is up to date]
    - Downloading remotes from RSPM ...             OK [file is up to date]
    - Downloading withr from RSPM ...               OK [file is up to date]
    - Downloading docopt from RSPM ...              OK [file is up to date]
    - Downloading evaluate from RSPM ...            OK [file is up to date]
    - Downloading fs from RSPM ...                  OK [file is up to date]
    - Downloading glue from RSPM ...                OK [file is up to date]
    - Downloading here from RSPM ...                OK [file is up to date]
    - Downloading rprojroot from RSPM ...           OK [file is up to date]
    - Downloading highr from RSPM ...               OK [file is up to date]
    - Downloading xfun from RSPM ...                OK [file is up to date]
    - Downloading hunspell from RSPM ...            OK [file is up to date]
    - Downloading knitr from RSPM ...               OK [file is up to date]
    - Downloading yaml from RSPM ...                OK [file is up to date]
    - Downloading lazyeval from RSPM ...            OK [file is up to date]
    - Downloading lifecycle from RSPM ...           OK [file is up to date]
    - Downloading rlang from RSPM ...               OK [file is up to date]
    - Downloading lintr from RSPM ...               OK [file is up to date]
    - Downloading rex from RSPM ...                 OK [file is up to date]
    - Downloading xml2 from RSPM ...                OK [file is up to date]
    - Downloading xmlparsedata from RSPM ...        OK [file is up to date]
    - Downloading magrittr from RSPM ...            OK [file is up to date]
    - Downloading pkgbuild from RSPM ...            OK [file is up to date]
    - Downloading pkgload from RSPM ...             OK [file is up to date]
    - Downloading purrr from RSPM ...               OK [file is up to date]
    - Downloading vctrs from RSPM ...               OK [file is up to date]
    - Downloading roxygen2 from RSPM ...            OK [file is up to date]
    - Downloading stringi from RSPM ...             OK [file is up to date]
    - Downloading stringr from RSPM ...             OK [file is up to date]
    - Downloading spelling from RSPM ...            OK [file is up to date]
    - Downloading styler from RSPM ...              OK [file is up to date]
    Successfully downloaded 45 packages in 55 seconds.

    # Installing packages --------------------------------------------------------
    - Installing R.methodsS3 ...                    OK [linked from cache]
    - Installing R.oo ...                           OK [linked from cache]
    - Installing R.utils ...                        OK [linked from cache]
    - Installing digest ...                         FAILED
    Traceback (most recent calls last):
    13: renv::restore()
    12: renv_restore_run_actions(project, diff, current, lockfile, rebuild)
    11: renv_install_impl(records)
    10: renv_install_staged(records)
     9: renv_install_default(records)
     8: handler(package, renv_install_package(record))
     7: renv_install_package(record)
     6: withCallingHandlers(renv_install_package_impl(record), error = function(e) writef("FAILED"))
     5: renv_install_package_impl(record)
     4: r_cmd_install(package, path)
     3: r_exec_error(package, output, "install", status)
     2: abort(all)
     1: stop(fallback)
stderr:
    Error: Error installing package 'digest':
    ==================================

    * installing *source* package 'digest' ...
    ** this is package 'digest' version '0.6.36'
    ** package 'digest' successfully unpacked and MD5 sums checked
    ** using staged installation
    ** libs
    using C compiler: 'gcc.exe (GCC) 14.2.0'
    using C++ compiler: 'G__~1.EXE (GCC) 14.2.0'
    g++ -std=gnu++17  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign    -c SpookyV2.cpp -o SpookyV2.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c aes.c -o aes.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c blake3.c -o blake3.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c blake3_dispatch.c -o blake3_dispatch.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c blake3_portable.c -o blake3_portable.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c crc32.c -o crc32.o
    g++ -std=gnu++17  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign    -c crc32c.cpp -o crc32c.o
    g++ -std=gnu++17  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign    -c crc32c_portable.cpp -o crc32c_portable.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c digest.c -o digest.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c digest2int.c -o digest2int.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c init.c -o init.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c md5.c -o md5.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c pmurhash.c -o pmurhash.o
    gcc  -I"C:/PROGRA~1/R/R-45~1.1/include" -DNDEBUG -I.    -I"C:/rtools45/x86_64-w64-mingw32.static.posix/include"      -O2 -Wall -std=gnu2x  -mfpmath=sse -msse2 -mstackrealign   -c raes.c -o raes.o
    raes.c: In function 'AESFinalizer':
    raes.c:25:3: error: implicit declaration of function 'Free'; did you mean 'free'? [-Wimplicit-function-declaration]
       25 |   Free(ctx);
          |   ^~~~
          |   free
    raes.c: In function 'AESinit':
    raes.c:42:23: error: implicit declaration of function 'Calloc'; did you mean 'calloc'? [-Wimplicit-function-declaration]
       42 |   ctx = (aes_context*)Calloc(sizeof(*ctx), char);
          |                       ^~~~~~
          |                       calloc
    raes.c:42:44: error: expected expression before 'char'
       42 |   ctx = (aes_context*)Calloc(sizeof(*ctx), char);
          |                                            ^~~~
    make: *** [C:/PROGRA~1/R/R-45~1.1/etc/x64/Makeconf:289: raes.o] Error 1
    ERROR: compilation failed for package 'digest'
    * removing 'C:/Users/User/.cache/pre-commit/repo_een0mgj/renv-default/renv/staging/1/digest'
    install of package 'digest' failed [error code 1]
    Execution halted
Check the log at C:\Users\User\.cache\pre-commit\pre-commit.log
(ve-data-science-py3.12) 
```

nickwctan commented:
> @arne-exe We also have the `ve_data_science` website here:
> 
> https://imperialcollegelondon.github.io/ve_data_science/getting_started/
> 
Sorry for budging in this PR, but Arne and I found out that the [Getting Started](https://github.com/ImperialCollegeLondon/ve_data_science/wiki/Getting-started-with-the-repository) from the `ve_data_science`
`/README.md` does not work anymore. It seems like the ending url is different, so maybe README.md file just needs updating? 

Same for the link to [GitHub overview](https://github.com/ImperialCollegeLondon/ve_data_science/wiki/Github-Overview) 

I have created a PR to change the links. See PR #126

arne-exe committed:
Changes to Dobert PFT species classification script

arne-exe committed:
Merge branch 'main' into 117-change-how-trees-are-assigned-into-pfts

arne-exe committed:
Merge after precommit issues
Merge branch '117-change-how-trees-are-assigned-into-pfts' of https://github.com/ImperialCollegeLondon/ve_data_science into 117-change-how-trees-are-assigned-into-pfts

arne-exe committed:
Added Dobert PFT cohort distribution script

arne-exe committed:
Check and clean up PFT base species classification script

arne-exe committed:
Checked and cleaned t_model_parameters script

arne-exe committed:
Updated csv which is still being tracked in the repo

arne-exe committed:
update and clean PFT species classification maxheight script and csv

arne-exe commented:
The base cohort distribution script "corrects" for trees with "unknown" PFT by distributing them evenly across the existing PFTs. However, it does not correct for trees that were excluded due to not having DBH measurements. So, the correction for trees with unknown PFT is only partial. It may be more accurate to not correct for unknown PFT in the cohort distribution script, and thus only focus on trees with known PFTs, but then correct the entire cohort distribution later on (i.e., as part of preparing site-specific input data) by working with stem density.

arne-exe committed:
Check and clean cohort distribution script and csvs

arne-exe committed:
Small correction height threshold

arne-exe commented:
> The base cohort distribution script "corrects" for trees with "unknown" PFT by distributing them evenly across the existing PFTs. However, it does not correct for trees that were excluded due to not having DBH measurements. So, the correction for trees with unknown PFT is only partial. It may be more accurate to not correct for unknown PFT in the cohort distribution script, and thus only focus on trees with known PFTs, but then correct the entire cohort distribution later on (i.e., as part of preparing site-specific input data) by working with stem density.

The same likely goes for the "unknown" PFT individuals in data/derived/plant/plant_functional_type/plant_functional_type_cohort_distribution_dobert which are "corrected" for in the same way

arne-exe commented:
Hi @annarallings or @davidorme, the new PFTs based on Dobert is something we'll need to discuss in more detail (i.e., the names and complexity may change depending on what is desired from the animal/plant models and their interaction), but I think this PR is now ready to be merged. Could either of you please have a look and approve?

arne-exe commented:
> Hi @annarallings or @davidorme, the new PFTs based on Dobert is something we'll need to discuss in more detail (i.e., the names and complexity may change depending on what is desired from the animal/plant models and their interaction), but I think this PR is now ready to be merged. Could either of you please have a look and approve?

Hi @hrlai, if you have time, could you please have a look at and approve this PR so we can get it merged?

arne-exe committed:
minor correction to input path

annarallings reviewed (approved):
LGTM  great update, Arne

arne-exe committed:
Merge branch 'main' into 117-change-how-trees-are-assigned-into-pfts
